onadata/libs/utils/logger_tools.py

```diff
@@ -336,6 +336,10 @@ def save_attachments(xform, instance, media_files, remove_deleted_media=False):
             xform.instances_with_osm = True
             xform.save()
         filename = os.path.basename(f.name)
+        # Validate Attachment file name length
+        if len(filename) > 100:
+            raise AttachmentNameError(
+                _('Attachment file name exceeds 100 characters'))
```

Comment:
Let's ensure this exception is caught and an appropriate status code is returned with the reason.